### PR TITLE
Fix non GitHub internal build

### PR DESCRIPTION
**Pull request by mayuki**
I tried to build for debugging. But the build has failed with one error.
It seems that caused by settings for `Internal` build on GitHub.

So I fixed assembly signing of `GitHub.InlineReview.csproj`. It's enabled by `Internal` build only.

- See AppVeyor build log: https://ci.appveyor.com/project/github-windows/visualstudio/build/2944

And I had encountered another error when I was debugging PR #1100.
IMetricsService is exposed only `Internal` build but UsageTracker requires that.

### CODE REVIEW

**Comment by shana** [comment]
Hey @mayuki, thanks for the patch! That reference to the internal signing key is indeed a bug, good catch.

The `NullMetricsService` class should not be needed, on non-internal builds any `IUsageTracker` MEF arguments are expected to be null. What error did you get for that?

**Comment by mayuki** [comment]
Hi, @shana, I got `InvalidOperationException` and a message `"Could not locate any instances of contract GitHub.Services.IMetricsService."`.

[UsageTracker uses GitHubServiceProvider](https://github.com/github/VisualStudio/blob/a6b9482be4982a245e476f2e316f3a7ae5598802/src/GitHub.VisualStudio/Services/UsageTracker.cs#L213) and [GitHubServiceProvider.GetService throws an exception](https://github.com/github/VisualStudio/blob/a6b9482be4982a245e476f2e316f3a7ae5598802/src/GitHub.VisualStudio/Services/UsageTracker.cs#L213) when can't resolve the class.
<details>
<summary>Stacktrace</summary>

```
   at GitHub.VisualStudio.GitHubServiceProvider.GetService(Type serviceType) in C:\Users\Tomoyo\Documents\Projects\github-VisualStudio\src\GitHub.VisualStudio\Services\GitHubServiceProvider.cs:line 212
   at GitHub.VisualStudio.GitHubServiceProvider.GetService[T]() in C:\Users\Tomoyo\Documents\Projects\github-VisualStudio\src\GitHub.VisualStudio\Services\GitHubServiceProvider.cs:line 218
   at GitHub.Services.UsageTracker.<Initialize>d__32.MoveNext() in C:\Users\Tomoyo\Documents\Projects\github-VisualStudio\src\GitHub.VisualStudio\Services\UsageTracker.cs:line 213
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at GitHub.Services.UsageTracker.<LoadUsage>d__33.MoveNext() in C:\Users\Tomoyo\Documents\Projects\github-VisualStudio\src\GitHub.VisualStudio\Services\UsageTracker.cs:line 228
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at GitHub.Services.UsageTracker.<IncrementOpenInGitHubCount>d__20.MoveNext() in C:\Users\Tomoyo\Documents\Projects\github-VisualStudio\src\GitHub.VisualStudio\Services\UsageTracker.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at GitHub.VisualStudio.Menus.OpenLink.<Activate>d__5.MoveNext() in C:\Users\Tomoyo\Documents\Projects\github-VisualStudio\src\GitHub.VisualStudio\Menus\OpenLink.cs:line 31
```

</details>

**Comment by shana** [comment]
> Hi, @shana, I got InvalidOperationException and a message "Could not locate any instances of contract GitHub.Services.IMetricsService.".

@mayuki Aaah, that is indeed a bug, it is supposed to fallback to not using it, it shouldn't throw. I guess we haven't been properly testing public builds as thoroughly as we should have been lately. Your solution of adding a dummy service might be better though 👍

File: src/GitHub.Exports/GitHub.Exports.csproj [hunk 1]
```diff
@@ -142,6 +142,7 @@
     <Compile Include="Models\ICommentModel.cs" />
     <Compile Include="Models\IInlineCommentModel.cs" />
     <Compile Include="Models\IPullRequestReviewCommentModel.cs" />
+    <Compile Include="Services\NullMetricsService.cs" Condition="$(Buildtype) != 'Internal'" />
```
Comment (shana) [inline src/GitHub.Exports/GitHub.Exports.csproj]: The check here should include `'` for safety (msbuild being all finicky about things at times. Like 

`Condition="'$(BuildType)' != 'Internal'"`

File: src/GitHub.InlineReviews/GitHub.InlineReviews.csproj [hunk 2]
```diff
@@ -15,7 +15,7 @@
   <PropertyGroup>
     <SignAssembly>true</SignAssembly>
   </PropertyGroup>
-  <PropertyGroup>
+  <PropertyGroup Condition="$(Buildtype) == 'Internal'">
```
Comment (shana) [inline src/GitHub.InlineReviews/GitHub.InlineReviews.csproj]: Same as above, the check here should include `'` for safety

`Condition="'$(BuildType)' != 'Internal'"`

**Comment by mayuki** [comment]
@shana, thank you for feedback. I had misunderstood [UsageTracker.SendUsage](https://github.com/github/VisualStudio/blob/master/src/GitHub.VisualStudio/Services/UsageTracker.cs#L308) throws an exception when a client is null. However, it doesn't. (if the client is null, `UsageTracker` disposes of a send timer) . 
so, I fixed to use `TryGetService` instead of `GetService` which solution is better than the first my solution. 😃

**Comment by shana** [comment]
@mayuki This is great, thank you so much for the patch and the patience!

**Pull request merged**